whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/FullTextSearchService.java

```diff
@@ -65,6 +68,23 @@ public Response search(
         }
     }
 
+    private static String escapeColon(final String query){
+        if (StringUtils.isEmpty(query) || !query.contains(":") || query.contains("\\:")){
+            return query;
+        }
+
+        final StringBuilder sb = new StringBuilder();
+
+        final String[] split = query.split(":");
+        Arrays.stream(split).limit(split.length - 1).forEach(splitColon -> sb.append(splitColon).append(shouldEscapeColon(splitColon) ? "\\:" : ":"));
+        return sb.append(split[split.length - 1]).toString();
+    }
+
+    private static boolean shouldEscapeColon(final String splitColon) {
+        final String[] words = splitColon.split("[^\\w-]+"); //Split in words (including dashes)
+        return words.length == 0 || (AttributeType.getByNameOrNull(words[words.length - 1]) == null && !"object-type" .equals(words[words.length - 1]));
+    }
```

Comment:
Use OBJECT_TYPE_FIELD_NAME ? Also since AttributeType using ciString equals may be use equalsIgnoreIcase for object-type also or convert it to cistring? Sorry did not check while refactoring

Reply:
Done